pages/cadastro/recuperar/index.public.js

```diff
@@ -100,27 +100,62 @@ function RecoverPasswordForm() {
       <Box sx={{ display: 'flex', flexDirection: 'column', gap: 3 }}>
         {globalErrorMessage && <Flash variant="danger">{globalErrorMessage}</Flash>}
 
-        <FormControl id="username">
-          <FormControl.Label>Digite seu usuário ou e-mail</FormControl.Label>
-          <TextInput
-            ref={userInputRef}
-            onChange={clearErrors}
-            name="userInput"
-            size="large"
-            autoCorrect="off"
-            autoCapitalize="off"
-            spellCheck={false}
-            block={true}
-            aria-label="Seu usuário ou e-mail"
-          />
-          {['userInput', 'email', 'username'].includes(errorObject?.key) && (
-            <FormControl.Validation variant="error">{errorObject.message}</FormControl.Validation>
-          )}
-
-          {errorObject?.type === 'string.alphanum' && (
-            <FormControl.Caption>Dica: use somente letras e números, por exemplo: nomeSobrenome4 </FormControl.Caption>
-          )}
-        </FormControl>
+        {user?.features.includes('create:recovery_token:username') && (
+          <Flash variant="default">
+            Você pode ajudar outra pessoa a recuperar sua senha, é só digitar o nome de usuário dela.
+          </Flash>
+        )}
+
+        {user?.features.includes('create:recovery_token:username') && (
+          <FormControl id="userInput">
+            <FormControl.Label>Digite seu e-mail ou o nome de usuário da pessoa que deseja ajudar</FormControl.Label>
+            <TextInput
+              ref={userInputRef}
+              onChange={clearErrors}
+              name="userInput"
+              size="large"
+              autoCorrect="off"
+              autoCapitalize="off"
+              spellCheck={false}
+              block={true}
+              aria-label="Digite seu e-mail ou o nome de usuário de outra pessoa"
+            />
+            {['userInput', 'email', 'username'].includes(errorObject?.key) && (
+              <FormControl.Validation variant="error">{errorObject.message}</FormControl.Validation>
+            )}
+
+            {errorObject?.type === 'string.alphanum' && (
+              <FormControl.Caption>
+                Dica: use somente letras e números, por exemplo: nomeSobrenome4{' '}
+              </FormControl.Caption>
+            )}
+          </FormControl>
+        )}
+
+        {!user?.features.includes('create:recovery_token:username') && (
+          <FormControl id="userInput">
+            <FormControl.Label>Digite seu e-mail</FormControl.Label>
+            <TextInput
+              ref={userInputRef}
+              onChange={clearErrors}
+              name="userInput"
+              size="large"
+              autoCorrect="off"
+              autoCapitalize="off"
+              spellCheck={false}
+              block={true}
+              aria-label="Seu e-mail"
+            />
+            {['userInput', 'email'].includes(errorObject?.key) && (
```

Comment:
```suggestion
            {['userInput', 'email', 'username'].includes(errorObject?.key) && (
```

Precisa aqui também, pois se o usuário digitar um `username` reservado (admin, user etc.) ele tem que ver a mensagem de erro.

Reply:
Show! Muito bem lembrado, havia me esquecido do array de `reservedUsernames`.

Vou alterar :+1:

Reply:
Feito :handshake: